Project-Descriptions-and-Plans/LLVM-Toolchain/COREV_LLVM_Plan_Approved_2023.md

```diff
@@ -0,0 +1,173 @@
+# OpenHW Project: corev-llvm-project Plan Approved CV32E40P-V2
+
+| Gate                                 | Status                                                     |
+| ------------------------------------ | ---------------------------------------------------------- |
+| PC gate: Project Concept             | Approved on 2020-11-21 as Preliminary Project Launch (PPL) |
+| PL gate: Project Launch              | Approved on 2022-xx-x                                     |
+| PA gate: Plan Approved               | Presented on 2022-11-28 for **2023 workplan**, approved on 2023-xx-xx |
+| PA gate: Plan Approved               | 2024 and beyond workplan in future PA gate |
+
+Author: ChunyuLiao
+
+## Summary of project
+
+This document should be read in the context of the overall CORE-V LLVM project project launch proposal.  It does not duplicate information from that document.
+
+This plan is to deliver Clang/LLVM compiler support for all of the CV32E40P version 2 instruction set extensions.
+
+### Summary of Timeline
+
+CORE-V LLVM project is a long-run project. It is therefore difficult to plan the complete project.
+
+Therefore, in order to keep the project under control, the project plan is defined for 2023 activity. 
+
+## OpenHW Members/Participants committed to participate in CORE-V LLVM project
+
+- Embecosm 
+- Institute of Software Chinese Academy of Sciences (ISCAS/PLCT)
+
+## Project Leader(s)
+
+- Charlie Keaney (overall LLVM project lead)
+- Chunyu Liao (project lead for this project)
+
+## Project Planning Documents
+
+The progress towards the milestones will be tracked during progress meetings, usually every week.
+
+A monthly report will be provided to the Software Task Group.
+
+## Summary of requirements
+
+### CORE-V LLVM project features
+
+After approval by the TWG, it will be managed in OpenHW GitHub.
+
+As a summary, contains:
+
++ Clang/LLVM tool chain for C and C++:
+    + assembler
+    + compiler
+    + support for ISA extensions
+
+The project will use the existing GNU linker and debugger. The standard RISC-V newlib and compilerRT libraries will be used unchanged.
+
+### License scheme
+
+The LLVM project mostly uses Apache licensing. The full details of all licenses is addressed in the Project Launch document.
+
+### Verification
+
+Github Continuous Integration and Test (CI) will be set up.
```

Comment:
Can you specify the CI we need?

Reply:
This will be GitHub CI to run the LLVM Integration Tests (_lit_) and in time a subset of the GCC regression test suite (needed to test execution of LLVM for embedded platforms).

Reply:
Initial support for CI: https://github.com/openhwgroup/corev-llvm-project/pull/23